armi/reactor/components/basicShapes.py

```diff
@@ -149,27 +151,14 @@ def getCircleInnerDiameter(self, Tc=None, cold=False):
         return 2.0 * sideLength
 
     def getComponentArea(self, cold=False, Tc=None):
-        """
-        Computes the area for the hexagon component in cm^2.
-
-        Notes
-        -----
-        http://www3.wolframalpha.com/input/?i=hexagon
-        """
+        """Computes the area for the hexagon component in cm^2."""
         op = self.getDimension("op", cold=cold, Tc=Tc)
         ip = self.getDimension("ip", cold=cold, Tc=Tc)
         mult = self.getDimension("mult")
         area = math.sqrt(3.0) / 2.0 * (op**2 - ip**2)
         area *= mult
         return area
 
-    def getPerimeter(self, Tc=None):
-        """Computes the perimeter of the hexagon component in cm."""
-        ip = self.getDimension("ip", Tc)
-        mult = self.getDimension("mult", Tc)
-        perimeter = 6 * (ip / math.sqrt(3)) * mult
-        return perimeter
```

Comment:
Why is this being deleted?

Reply:
It turns out, no one is using perimeter on a hex shape, I checked.

Reply:
I don't see how that is related to the purpose of this PR, though. 

I'm not trying to be provocative, this just seems a bit random and unwarranted unless there is something actually wrong with the perimeter calculation. Maybe there is a reason that I am not aware of for removing stuff like this.

Reply:
I don't know if this calculation is correct. Should the perimeter of a hexagon with a hexagonal hole cut out of it include the internal perimeter AND the external one?

Well, for SOME reason, this only includes the internal and not the external perimeter. Why is that? What does that mean?

At the very least, this method needs a long docstring telling me what it is for. But it doesn't have that, so I can only guess.

But, @keckler , to answer your question, I am subclassing this class. And thus would need to subclass this method and re-implement it. But how can I reimplement something whose purpose I don't understand. That's how I got to this point.

Thoughts?

Reply:
Makes sense, thank you both for the context.